File: 347-Top-K-Frequent-Elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int[] res = new int[k];
        Map<Integer, Integer> freqCount = new HashMap<>();

        for(int num =0; num < nums.length; num++){
            freqCount.put(nums[num],freqCount.getOrDefault(nums[num],0)+1);
            //minHeap.offer(nums[num]);
        }
        PriorityQueue<Integer> minHeap = new PriorityQueue<>(
            (a,b) -> freqCount.get(b)-freqCount.get(a));
        
        minHeap.addAll(freqCount.keySet());

        for(int resCount =0 ; resCount<k; resCount++){
            res[resCount] = minHeap.poll();
        }
        return res;
    }
}

File: 347-Top-K-Frequent-Elements.py
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counted = {}
        pq =[]
        for val in range(len(nums)):
            counted[nums[val]] = 1+ counted.get(nums[val],0)
        for key, val in counted.items():
            heapq.heappush(pq,(-val,key))
        res= []
        for count in range(k):
            res.append(heapq.heappop(pq)[1])
        return res
